test/Microsoft.Health.Fhir.Tests.E2E.Shared/FhirClient.cs

```diff
@@ -0,0 +1,565 @@
+                new KeyValuePair<string, string>("scope", AuthenticationSettings.Scope),
+                new KeyValuePair<string, string>("resource", AuthenticationSettings.Resource),
+                new KeyValuePair<string, string>("username", user.UserId),
+                new KeyValuePair<string, string>("password", user.Password),
+            };
+        }
+
+        public async Task GetSecuritySettings(string fhirServerMetadataUrl)
+        {
+            FhirResponse<ResourceElement> readResponse = await ReadAsync(fhirServerMetadataUrl);
+
+            var metadata = readResponse.Resource.ToPoco<CapabilityStatement>();
+
+            foreach (var rest in metadata.Rest.Where(r => r.Mode == CapabilityStatement.RestfulCapabilityMode.Server))
+            {
+                var oauth = rest.Security?.GetExtension(Constants.SmartOAuthUriExtension);
+                if (oauth != null)
+                {
+                    var tokenUrl = oauth.GetExtensionValue<FhirUri>(Constants.SmartOAuthUriExtensionToken).Value;
+                    var authorizeUrl = oauth.GetExtensionValue<FhirUri>(Constants.SmartOAuthUriExtensionAuthorize).Value;
+
+                    SecuritySettings = (true, authorizeUrl, tokenUrl);
+                    return;
+                }
+            }
+
+            SecuritySettings = (false, null, null);
+        }
+
+        public ResourceElement GetDefaultObservation()
+        {
+            return GetJsonSample("Weight");
+        }
+
+        public ResourceElement GetDefaultPatient()
+        {
+            return GetJsonSample("Patient");
+        }
+
+        public ResourceElement GetDefaultOrganization()
+        {
+            return GetJsonSample("Organization");
+        }
+
+        public ResourceElement GetEmptyRiskAssessment()
+        {
+            return new RiskAssessment().ToResourceElement();
+        }
+
+        public ResourceElement GetEmptyObservation()
+        {
+            return new Observation().ToResourceElement();
+        }
+
+        public ResourceElement GetEmptyValueSet()
+        {
+            return new ValueSet().ToResourceElement();
+        }
+
+        /// <summary>
+        /// Gets back a resource from a json sample file.
+        /// </summary>
+        /// <param name="fileName">The JSON filename, omit the extension</param>
+        public ResourceElement GetJsonSample(string fileName)
+        {
+            var json = EmbeddedResourceManager.GetStringContent("TestFiles", fileName, "json", _fhirVersion.ToString());
+            var parser = new Hl7.Fhir.Serialization.FhirJsonParser();
+            return parser.Parse<Resource>(json).ToResourceElement();
+        }
+
+        public void Validate(ResourceElement resourceElement)
+        {
+            DotNetAttributeValidation.Validate(resourceElement.ToPoco(), true);
+        }
+
+        public ResourceElement UpdateId(ResourceElement resourceElement, string id)
```

Comment:
nit: maybe make this file partial and separate the implementation of the "operations" and "test extensions"?